File: languages/django_models/law.py
from .abstract_element_model import AbstractElementModel
from django.db import models

class Law(AbstractElementModel):

    # Code
    declaration = models.TextField(blank=True, null=True)
    purpose = models.TextField(blank=True, null=True)
    date = models.PositiveIntegerField(blank=True, null=True)
    parent_law = models.ForeignKey("Law", on_delete=models.SET_NULL, blank=True, null=True, related_name="law_parent_law")
    penalties = models.ManyToManyField("Construct", blank=True, related_name="law_penalties")

    # World
    author = models.ForeignKey("Institution", on_delete=models.SET_NULL, blank=True, null=True, related_name="law_author")
    locations = models.ManyToManyField("Location", blank=True, related_name="law_locations")
    zones = models.ManyToManyField("Zone", blank=True, related_name="law_zones")
    prohibitions = models.ManyToManyField("Construct", blank=True, related_name="law_prohibitions")
    adjudicators = models.ManyToManyField("Title", blank=True, related_name="law_adjudicators")
    enforcers = models.ManyToManyField("Title", blank=True, related_name="law_enforcers")

    def __str__(self):
        return self.name

File: languages/csharp_unity_json_properties/Law.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Utils;

[System.Serializable]
public class Law : Element
{
    [JsonProperty("declaration"), TextAttribute("")]
    public string declaration;
    [JsonProperty("purpose"), TextAttribute("")]
    public string purpose;
    [JsonProperty("date"), Integer(0)]
    public int date;
    [JsonProperty("parent_law"), ReferenceAttribute(typeof(Law))]
    public string parentLaw;
    [JsonProperty("penalties"), ReferenceAttribute(typeof(Construct), true)]
    public string penalties;
    [JsonProperty("author"), ReferenceAttribute(typeof(Institution))]
    public string author;
    [JsonProperty("locations"), ReferenceAttribute(typeof(Location), true)]
    public string locations;
    [JsonProperty("zones"), ReferenceAttribute(typeof(Zone), true)]
    public string zones;
    [JsonProperty("prohibitions"), ReferenceAttribute(typeof(Construct), true)]
    public string prohibitions;
    [JsonProperty("adjudicators"), ReferenceAttribute(typeof(Title), true)]
    public string adjudicators;
    [JsonProperty("enforcers"), ReferenceAttribute(typeof(Title), true)]
    public string enforcers;
}
